index.html

```diff
@@ -1,45 +1,451 @@
+                    a stable and private id for the device, <var>deviceId</var>,
+                    set [[\devicesLiveMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already <code>true</code>,
+                    and set the
+                    [[\devicesAccessibleMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already
+                    <code>true</code>.
+                  </p>
+                  <p>
+                    The user agent MUST NOT store a {{PermissionState/"granted"}} permission entry.
+                  </p>
+                  <p>If the result is {{PermissionState/"denied"}}, jump to the step labeled
+                    <em>Permission Failure</em> below. If the user never
+                    responds, this algorithm stalls on this step.
+                  </p>
+                  <p>If the user grants permission but a hardware error
+                    such as an OS/program/webpage lock prevents access to at least one device,
+                    <a>reject</a> <var>p</var> with a new
+                    {{DOMException}} object whose
+                    {{DOMException/name}} attribute has the value
+                    {{NotReadableError}} and abort these steps.
+                  </p>
+                  <p>If the result is {{PermissionState/"granted"}} but device access fails for
+                    any reason other than those listed above, <a>reject</a>
+                    <var>p</var> with a new {{DOMException}}
+                    object whose {{DOMException/name}} attribute has the
+                    value {{AbortError}} and abort these steps.
+                  </p>
+                </li>
+                <li>
+                  <p>Let <var>streams</var> be the
+                    list of {{MediaStream}} objects for which the user granted permission.</p>
+                </li>
+                <li>
+                  <!-- Guidance taken from https://infra.spec.whatwg.org/ -->
+                  <p>For each <var>stream</var> of <var>streams</var>
+                    run the [=ApplyConstraints algorithm=] on all
+                    tracks in <var>stream</var> with the appropriate
+                    constraints. Should this fail for at least one <var>stream</var>,
+                    let <var>failedConstraint</var>
+                    be the result of the algorithm that failed, and let
+                    <var>message</var> be either <code>undefined</code> or an
+                    informative human-readable message, then <a>reject</a>
+                    <var>p</var> with a new <code>OverconstrainedError</code>
+                    created by calling
+                    <code>OverconstrainedError(<var>failedConstraint</var>,
+                    <var>message</var>)</code>, and then abort the processing
+                    of all other elements <var>stream'</var> in <var>streams</var>.
+                  </p>
```

Comment:
Another section we could drop if we skip having to apply constraints immediately when the capture starts.

Reply:
It's gone.